lib/src/platform.dart

```diff
@@ -111,49 +111,106 @@ PlatformTarget platform(BuildContext context) {
   }
 }
 
+abstract class _DialogBaseData {
+  final WidgetBuilder? builder;
+  final bool? barrierDismissible;
+  final RouteSettings? routeSettings;
+  final bool? useRootNavigator;
+  final String? barrierLabel;
+  final Offset? anchorPoint;
+
+  _DialogBaseData({
+    this.builder,
+    this.barrierDismissible,
+    this.routeSettings,
+    this.useRootNavigator,
+    this.barrierLabel,
+    this.anchorPoint,
+  });
+}
+
+class MaterialDialogData extends _DialogBaseData {
+  final bool? useSafeArea;
+  final Color? barrierColor;
+
+  MaterialDialogData({
+    super.builder,
+    super.barrierDismissible,
+    super.routeSettings,
+    super.useRootNavigator,
+    super.barrierLabel,
+    super.anchorPoint,
+    this.useSafeArea,
+    this.barrierColor,
+  });
+}
+
+class CupertinoDialogData extends _DialogBaseData {
+  CupertinoDialogData({
+    super.builder,
+    super.barrierDismissible,
+    super.routeSettings,
+    super.useRootNavigator,
+    super.barrierLabel,
+    super.anchorPoint,
+  });
+}
+
 Future<T?> showPlatformDialog<T>({
   required BuildContext context,
-  required WidgetBuilder builder,
+  MaterialDialogData? material,
+  CupertinoDialogData? cupertino,
+  WidgetBuilder? builder,
   bool? barrierDismissible,
   RouteSettings? routeSettings,
   bool useRootNavigator = true,
-  bool materialUseSafeArea = true,
-  Color? materialBarrierColor = Colors.black54,
+  @Deprecated('Use material.useSafeArea instead')
+      bool materialUseSafeArea = true,
+  @Deprecated('Use material.barrierColor instead')
+      Color? materialBarrierColor = Colors.black54,
   String? barrierLabel,
+  Offset? anchorPoint,
 }) {
   if (isMaterial(context)) {
+    assert(material?.builder != null || builder != null);
+
     return showDialog<T>(
       context: context,
-      builder: builder,
-      barrierDismissible: barrierDismissible ?? true,
-      routeSettings: routeSettings,
-      useRootNavigator: useRootNavigator,
-      useSafeArea: materialUseSafeArea,
-      //child: , deprecated
-      barrierColor: materialBarrierColor,
-      barrierLabel: barrierLabel,
+      builder: material?.builder ?? builder!,
+      barrierDismissible:
+          material?.barrierDismissible ?? barrierDismissible ?? true,
+      routeSettings: material?.routeSettings ?? routeSettings,
+      useRootNavigator: material?.useRootNavigator ?? useRootNavigator,
+      useSafeArea: material?.useSafeArea ?? materialUseSafeArea,
+      barrierColor: material?.barrierColor ?? materialBarrierColor,
+      barrierLabel: material?.barrierLabel ?? barrierLabel,
+      anchorPoint: material?.anchorPoint ?? anchorPoint,
     );
   } else {
+    assert(cupertino?.builder != null || cupertino != null);
```

Comment:
@aqwert `cupertino != null` is wrong and now crashes - needs to be `builder != null`